docs/how-to-build-with-ddn/with-mongodb.mdx

```diff
@@ -0,0 +1,397 @@
+
+:::tip
+
+The following steps are necessary each time you make changes to your **source** schema. This includes, adding,
+modifying, or dropping collections.
+
+:::
+
+#### Step 12.1. Re-introspect your data source
+
+```sh title="Run the introspection command again:"
+ddn connector introspect my_mongo
+```
+
+In `app/connector/my_mongo/configuration.json`, you'll see schema updated to include operations for the `posts`
+collection. You'll also see a `posts.json` file in the `schema` directory.
+
+#### Step 12.2. Update your metadata
+
+```sh title="Add the posts model:"
+ddn model add my_mongo posts
+```
+
+#### Step 12.3. Kill your services
+
+Bring down the services by pressing `CTRL+C` in the terminal tab logging their activity.
+
+#### Step 12.4. Create a new build
+
+```sh title="Next, create a new build:"
+ddn supergraph build local
+```
+
+#### Step 12.5 Restart your services
+
+```sh title="Bring everything back up:"
+ddn run docker-start
+```
+
+### Step 13. Query your new build
+
+```graphql title="Head back to your console and query the posts model:"
+query GetPosts {
+  posts {
+    userId
+    title
+    content
+  }
+}
+```
+
+```json title="You'll get a response like this:"
+{
+  "data": {
+    "posts": [
+      {
+        "userId": 1,
+        "title": "My First Post",
+        "content": "This is Alice's first post."
+      },
+      {
+        "userId": 1,
+        "title": "Another Post",
+        "content": "Alice writes again!"
+      },
+      {
+        "userId": 2,
+        "title": "Bob's Post",
+        "content": "Bob shares his thoughts."
+      },
+      {
+        "userId": 3,
+        "title": "Hello World",
+        "content": "Charlie joins the conversation."
+      }
+    ]
+  }
+}
+```
+
+### Step 14. Create a relationship
+
+```yaml title="Open the Posts.hml file and add the following to the end:"
+---
+kind: Relationship
+version: v1
+definition:
+  name: user
+  sourceType: Posts
+  target:
+    model:
+      name: Users
+      relationshipType: Object
+  mapping:
+    - source:
+        fieldPath:
+          - fieldName: userId
+      target:
+        modelField:
+          - fieldName: userId
+```
+
+:::tip LSP-Assisted authoring is available
+
+We've created an extension for VS Code that leverages LSP to make authoring these metadata objects easier. Check it out
+[here](https://marketplace.visualstudio.com/items?itemName=HasuraHQ.hasura).
+
+:::
+
+### Step 15. Rebuild your project
+
+Bring down the services by pressing `CTRL+C` in the terminal tab logging their activity.
+
+```sh title="As your metadata has changed, create a new build:"
+ddn supergraph build local
+```
+
+```sh title="Bring everything back up:"
+ddn run docker-start
+```
+
+### Step 16. Query using your relationship
+
+```graphql title="Now, execute a nested query using your relationship:"
+query GetPosts {
+  posts {
+    title
+    content
+    user {
+      userId
+      name
+      age
+    }
+  }
+}
+```
```

Comment:
Even with image delete and rebuild I'm getting 

```
app_my_mongo-1    | {"timestamp":"2024-12-04T13:16:10.646248Z","level":"INFO","fields":{"message":"Waiting for suitable server to become available","topologyId":"675055ff052b0c17ff39fe22","operation":"aggregate","selector":"ReadPreference { Mode: Primary }","topologyDescription":"{ Type: Unknown, Servers: [ { Address: local.hasura.dev:27017, Type: Unknown, Error: Kind: I/O error: Connection refused (os error 111), labels: {} } ] }","remainingTimeMS":"29999"},"target":"mongodb::server_selection","span":{"internal.visibility":"user","name":"MongoDB Aggregate Command"},"spans":[{"method":"POST","uri":"/query","version":"HTTP/1.1","name":"request"},{"internal.visibility":"user","name":"/query"},{"internal.visibility":"user","name":"Execute Query Pipeline"},{"internal.visibility":"user","name":"MongoDB Aggregate Command"}]}
app_my_mongo-1    | {"timestamp":"2024-12-04T13:16:18.802281Z","level":"ERROR","fields":{"error":"400 Bad Request: Kind: Server selection timeout: No available servers. Topology: { Type: Unknown, Servers: [ { Address: local.hasura.dev:27017, Type: Unknown, Error: Kind: I/O error: Connection refused (os error 111), labels: {} } ] }, labels: {}\n(details: {})"},"target":"mongodb_connector::mongo_connector","span":{"internal.visibility":"user","name":"/query"},"spans":[{"method":"POST","uri":"/query","version":"HTTP/1.1","name":"request"},{"internal.visibility":"user","name":"/query"}]}
```

With no error in Console, just empty response pane. 

🤷

Reply:
Oh wait I killed my mongo db. Ignore.